go/cmd/vtctldclient/command/routing_rules.go

```diff
@@ -82,7 +82,7 @@ func commandApplyRoutingRules(cmd *cobra.Command, args []string) error {
 	}
 
 	rr := &vschemapb.RoutingRules{}
-	if err := json2.Unmarshal(rulesBytes, &rr); err != nil {
+	if err := json2.Unmarshal(rulesBytes, rr); err != nil {
```

Comment:
@rohit-nayak-ps It looks like we have this kind of bug in a bunch more places, we should audit them all. 

I think the better fix here would be to have an explicit helper function that requires passing in the correct type so that it's not as easy to make this mistake.

Reply:
Same bug here:

https://github.com/vitessio/vitess/blob/6fb0f0e0f507e6e6e829783ae840fd4063eca9e2/go/cmd/vtctldclient/command/keyspace_routing_rules.go#L84-L87

https://github.com/vitessio/vitess/blob/6fb0f0e0f507e6e6e829783ae840fd4063eca9e2/go/cmd/vtctldclient/command/shard_routing_rules.go#L89-L92

Reply:
Added  `json2.UnmarshalPB()` to ensure static checks.